components/script/dom/notification.rs

```diff
@@ -83,7 +103,21 @@ pub(crate) struct Notification {
     require_interaction: bool,
     /// <https://notifications.spec.whatwg.org/#actions>
     actions: Vec<Action>,
-    // TODO: image resource, icon resource, and badge resource
+    /// Pending image, icon, badge, action icon resource request's id
+    #[no_trace] // RequestId is not traceable
+    pending_request_ids: DomRefCell<HashSet<RequestId>>,
+    /// <https://notifications.spec.whatwg.org/#image-resource>
+    #[ignore_malloc_size_of = "Arc"]
+    #[no_trace]
+    image_resource: DomRefCell<Option<Arc<Image>>>,
```

Comment:
Do other DOM objects also use Arc<Image> like this?

Reply:
There's one in Object element:
https://github.com/servo/servo/blob/f483a3d34b9f746d619c793af1f05ccbf0c84119/components/script/dom/htmlobjectelement.rs#L34

Image element ref the `ImageRequest`, which includes `Arc<Image>`:
https://github.com/servo/servo/blob/f483a3d34b9f746d619c793af1f05ccbf0c84119/components/script/dom/htmlimageelement.rs#L160
https://github.com/servo/servo/blob/f483a3d34b9f746d619c793af1f05ccbf0c84119/components/script/dom/htmlimageelement.rs#L149